eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -145,41 +147,49 @@ class Peer(nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: Actor
       goto(DISCONNECTED) using DisconnectedData(address_opt, channels)
 
     case Event(Terminated(actor), d@InitializingData(_, _, channels, _)) if channels.exists(_._2 == actor) =>
-      val h = channels.filter(_._2 == actor).map(_._1)
+      val h = channels.filter(_._2 == actor).keys
       log.info(s"channel closed: channelId=${h.mkString("/")}")
       stay using d.copy(channels = channels -- h)
   }
 
   when(CONNECTED, stateTimeout = nodeParams.pingInterval) {
-    case Event(StateTimeout, ConnectedData(_, transport, _, _, _, _)) =>
+    case Event(StateTimeout, d@ConnectedData(_, transport, _, _, _, _, _)) =>
       // no need to use secure random here
       val pingSize = Random.nextInt(1000)
       val pongSize = Random.nextInt(1000)
-      transport ! wire.Ping(pongSize, BinaryData("00" * pingSize))
-      stay
+      val ping = wire.Ping(pongSize, BinaryData("00" * pingSize))
+      val disconnector = context.system.scheduler.scheduleOnce(10 seconds, transport, PoisonPill)
```

Comment:
I would have created a dedicated class and have the scheduler send it instead.

```scala
 case class PingTimeout(ping: Ping)
```

That way you know exactly  which ping timed out, and you don't receive random `PoisonPill` that kill your connection.

```scala
case Event(PingTimeout(ping), ConnectedData(_, transport, _, _, _, _, _) =>
  log.warning(s"no response to ping=$ping, closing connection")
  transport ! PoisonPill
```